src/helper/runner.cpp

```diff
@@ -36,21 +37,50 @@ network::base_acceptor::ptr acceptor_from_config(const config::config &cfg)
 } // namespace
 
 runner::runner(const config::config &cfg)
-    : cfg_(cfg), engine_pool_{std::make_shared<engine_pool>()},
-      acceptor(acceptor_from_config(cfg))
-{
-}
+    : runner(cfg, acceptor_from_config(cfg)) {}
 
 runner::runner(const config::config &cfg, network::base_acceptor::ptr &&acceptor)
     : cfg_(cfg), engine_pool_{std::make_shared<engine_pool>()},
-      acceptor(std::move(acceptor))
+      acceptor_(std::move(acceptor)),
+      idle_timeout_(cfg.get<unsigned>("runner_idle_timeout"))
 {
+    try {
+        acceptor_->set_accept_timeout(1min);
+    } catch (const std::exception &e) {
+        // Not a critical error, we should continue
+        SPDLOG_WARN("Failed to set runner timeout: {}", e.what());
+    }
 }
 
 void runner::run() {
     try {
+        auto last_idle = std::chrono::steady_clock::now();
```

Comment:
This name is very misleading. It's actually the last time it was detected **not** to be idle.

Reply:
Fair point.